### DISCUSSION TITLE
Why check for login shell in the fish.config snippet ?

### DISCUSSION THREAD

**Discussion by HugoBelotDeloro** [pyenv/pyenv#2195] [Q&A] [answered]
In the installation instructions for the fish shell, it is recommended to add to the user's `config.fish` file the snippet:
```bash
status is-login; and pyenv init --path | source
status is-interactive; and pyenv init - | source
```
I do not understand the reason why `pyenv init --path | source` should only be run in login shells. I have researched what the difference is between login and non-login shell, and it appears most shells I run  (if not all) are non-login. However without running this command shims are not added to PATH and thus the python version I get when running python is still the system one.  
I can run this command manually, but I wonder why this check exists, especially since it there does not seem to be any such check for other shells. From my understanding, the login shells seem to be exactly the place where I do not want my python version to be changed in order to avoid messing with my other processes relying on python.

**Comment by native-api** [comment, marked as answer]
Any non-login shell is (in a normal setup) a descendant of a login shell.
If Fish is not your login shell, you need to add an equivalent command to your login shell's configuration.
In a Linux GUI session, the GUI manager acts as a Sh login shell (executing `~/.profile`).

Anyway, we're currently considering going forward with https://github.com/pyenv/pyenv/issues/2085 which will make a separate `--path` step unnecessary (except for some Bash setups).
